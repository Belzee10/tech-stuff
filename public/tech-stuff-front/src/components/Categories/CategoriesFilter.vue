<template>
  <div class="categories-filter">
    <v-checkbox
      v-for="item in categories"
      :key="item.id"
      v-model="value"
      :label="item.name"
    ></v-checkbox>
  </div>
</template>

<script>
export default {
  name: 'CategoriesFilter',
  props: {
    categories: {
      type: Array,
      default: () => []
    },
    value: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style lang="scss" scoped></style>
