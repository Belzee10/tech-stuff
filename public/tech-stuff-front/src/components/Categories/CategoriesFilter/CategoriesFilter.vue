<template>
  <div class="categories-filter pa-3">
    <v-radio-group :value="value" @change="value => $emit('check', value)">
      <v-radio
        v-for="item in categories"
        :key="item.id"
        color="primary"
        class="mb-2 category-check"
        :label="item.name"
        :value="item.id"
      ></v-radio>
    </v-radio-group>
  </div>
</template>

<script>
export default {
  name: 'CategoriesFilter',
  props: {
    categories: {
      type: Array,
      default: () => []
    },
    value: {
      type: [Number, String],
      default: 0
    }
  }
};
</script>

<style lang="scss"></style>
