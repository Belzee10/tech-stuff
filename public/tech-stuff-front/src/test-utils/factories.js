import { mount, createLocalVue } from '@vue/test-utils';
import mergeWith from 'lodash.mergewith';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

let vuetify;

beforeEach(() => {
  vuetify = new Vuetify();
});

const customizer = (_, srcValue) => {
  if (Array.isArray(srcValue)) {
    return srcValue;
  }
  if (srcValue instanceof Object && Object.keys(srcValue).length === 0) {
    return srcValue;
  }
};

/**
 * create component wrapper
 * @param {*} Component
 * @param {Object} overrides
 */
const createWrapper = (Component, overrides) => {
  const defaultMountingOptions = {
    localVue,
    vuetify
  };
  return mount(
    Component,
    mergeWith(defaultMountingOptions, overrides, customizer)
  );
};

/**
 * create Vuex store
 * @param {Object} overrides
 */
const createStore = overrides => {
  const defaultStoreConfig = {};
  return new Vuex.Store(mergeWith(defaultStoreConfig, overrides, customizer));
};

export { createWrapper, createStore };
