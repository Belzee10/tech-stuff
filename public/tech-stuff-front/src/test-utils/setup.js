import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.config.productionTip = false;

Vue.use(Vuetify);

/**
 * silent "console.error"
 */
global.console = {
  error: jest.fn(),
  log: console.log,
  warn: console.warn,
  info: console.info,
  debug: console.debug
};
