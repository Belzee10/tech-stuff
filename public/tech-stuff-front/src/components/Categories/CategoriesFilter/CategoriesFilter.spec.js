import { createWrapper } from '@/test-utils/factories';
import CategoriesFilter from './CategoriesFilter.vue';

describe('CategoriesFilter.vue', () => {
  const props = {
    categories: [{}]
  };
  test('should render all "categories"', () => {
    const wrapper = createWrapper(CategoriesFilter, {
      propsData: props
    });
    const categories = wrapper.findAll('.v-input');
    expect(categories).toHaveLength(1);
  });

  test('should render correctly', () => {
    const wrapper = createWrapper(CategoriesFilter, {
      propsData: props
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
