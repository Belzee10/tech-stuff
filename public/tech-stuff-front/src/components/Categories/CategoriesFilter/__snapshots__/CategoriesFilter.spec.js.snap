// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesFilter.vue should render correctly 1`] = `
<div
  class="categories-filter pa-3"
>
  <h3
    class="headline"
  >
    Categories
  </h3>
   
  <hr
    aria-orientation="horizontal"
    class="pb-5 v-divider theme--light"
    role="separator"
  />
   
  <div
    class="v-input ma-0 pa-0 v-input--dense theme--light v-input--selection-controls v-input--checkbox"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-input--selection-controls__input"
        >
          <input
            aria-checked="false"
            id="input-12"
            role="checkbox"
            type="checkbox"
            value=""
          />
          <div
            class="v-input--selection-controls__ripple"
          />
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"
          />
        </div>
      </div>
      <div
        class="v-messages theme--light"
      >
        <span
          class="v-messages__wrapper"
          name="message-transition"
          tag="div"
        />
      </div>
    </div>
  </div>
</div>
`;
